### ISSUE TITLE
Invalid root hash calculation

### ISSUE THREAD

**Issue by cytadela8** [aeternity/elixir-node#654]
Root hash calculation from tree hashes is invalid.

To reproduce:
channels root hash: `<<123, 61, 3, 117, 14, 251, 195, 47, 149, 121, 139, 42, 50, 2,116, 64, 114, 225, 142, 134, 254, 204, 23, 16, 80, 121, 135, 29, 110, 255,29, 84>>`
accounts root hash: `<<162, 234, 121, 135, 173, 98, 243, 71, 162, 50, 110, 120, 127, 50, 15, 21, 28, 63, 76, 27, 71, 149, 244, 201, 128, 9, 0, 195, 59, 158, 161, 254>>`
rest: empty

blocks to reproduce:
```
["CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutikLVdAqrqvqaquwLnGqTNoU8jy8ndyxwxsVTr2JJsRgFctDoC6Ccxfnq8fsCL2hsnWB4xshaBquFUb1WjhibGjwEqGK25kdQ5MeCZZH14fP6S1PkLa4h8trtnPhjzR8LgPrSZoHpjP7HW92q3K612zsQL8XqLyiMhX69dAHoxtJBKXg9MZaFWEuWPnTis7kuKMGmnhHN1YdjVYWKKpZynNyQ1rQ8fJzHczJ5jWbKTEhhp9D8P4uDLSLdBDUfvHqLEoym1HGapKedux4EXYKoojEEi5Z5VYnguayd7NUSY52ArFQLg1t28nxjbYD6jK6h2P73ztuGp6PNWpzmoifULoJsNnWwMprEuDB5jpn2BUrdg46NEEg3gVzo92epMz4Y1owPvSb4TTwbsWENWLqqz83tA949QUJkfULEHvNQdmFkLYcPL8xazrcf1LRRd",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutj1PVVse8sTrxd3Cf27tDYx9iNYpPnS5gB9csJYheAXTkqG7ECYJJynpDEaLEzMDs7fyb9UUFmb8XkZwuc3VmUJ3ZuMS7EMBqjFCu4tdgUGpuyNT5qyKgXo7QhqSFDaetXgjRHbo7XVtXyGUcYhSsDk2SPxtEMRWze4rvgyy5Rd7e9NWzuDyQXH2dGEPXQU8uRzJXikUHteCu8CZtsBYgSvBYfuxUj5E5CEmxi8HrajETaAjPEovPCKwEGCVxzygDjesUvcn1KRpi5zHVHNBy5QyLTj4XY494No7JjjkhtjLTAfKtRP3Pa4GpS31dVE3cZXYVCuhxqVdViHZvY1rdEDxrXB15HCukgYb59TajFt3x2mXsjxEwfLh78GqmgBBsDHdfpNsCHjuzZcin3bHTjoL5foF54YGaKqDcTmGKhufdwBYT4XKze4HKUWAKU",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutjJUKARiG64F5DUofdnSt9zHqEJ1HE5D6Pqi2hNLoFCJAnrMcJd1w52pC6B4Vuj8v6LQm2KXSP3qrtsovPHbRqxyagsG2mpkGBEeHEMC4rYVzHtgJKXcQ1hGHGtGdkUYF5NzjGRDVp998y7NFh2A8rv1skjjEvXzykMerJ2f9EoULo6zL3BWUyJY6r3ApiJtDKKRedEXE7PRrXdWf7FCiwTKop5DGbQrTR1Zn84E3F4pHAkXzYPWM8hvd55a6hYoM1B7ddwMieCMxac5L1tKhiD8hfRZy8QNBeGx3P3SGMbW9irVcpbBuowZBtKX22fiDz4saqU5MTbrn4KKYHTSjWznmmrBAqKWBuWi5wrwkz3dFdCfgLryxXPbWFTVaY5gH6giMKFqzXhquANfh8oxGfjQmCz1ZFp56AoCyX9uA2f1Qm61oPr6xiQu9Cpb53",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutjtJmV1c74QTSfaAG9rmTeqsLQVirDqhy2G16Ee4kkyVnoYKEbreWCYFHbhvzmbgjisGZ866yfcLfD3r1PfKxJMkVveVPProTAEJQwTenoa6sHDy4mjmck4qHPfDiWPtN6FR96isnE6WoSyMm1dSWebZmv9f6kSy9wfJT1W7yBQE9oYan9fGsXCtxN45f9xRwLqxBGLujr3i5EXXbBnm944z2SNPmtQijS2CyeTEBHxHNrKXCpi6GPe9aFMC6Dva3AQ7o3nS8iSyAF12g7vU98iLC35mzSCdjVACAqvPc8Uhk9RNFT2oM51UAcAxmWoxMCuH5zWbiKXsoLHaDxX6mmiMNA7MNkKfc4wSimTFQHU6ZqG1G7tdz7Y5MzPTsgzVY2zAyx1Cv1HvkK2zZN6DYuwPEgJGRqCdpHjsRoqAgoj5yieHT7t3kW8dQwZPLD",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutkWe34FRNr433X56Tes11u4LMXPFqeXe1R69Rd1MHEpBaHbTZgFPbwXeAPPGD3pMzao4aoRC7GiY7EHx1HSAVkU463nyzpjXN4K3eH1YKKhYHzCxxxoztkhhxBDH96PqWTV67avZRSfuEJaj6bZKdg7twYFShDsX4fJcMquCN6yhcb3j5k8eEz3ECctZYXXMSah9FghnYLYLhp4XdDvzhxh2YobRJXh9DQ1j2zcXkT9xnPL9Zsftr7RpQS13TisSMuEcUcCApqTHCxggrPf3FFKPjGfqkvU6E1ow8rvRcWYzGaX48rFaT7fzQxkAEbKthymugevskZnhfak4cfAxE1VXqniNjL46KHfpYdn7325ntiM7t43cNTLmDT9g76t2iGyjFQnxVVAPKSSNUzVc7aLHXFN1VszaBbpu8iarvWNy8WX1QfkCM4ELRDQYSA",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutmCr7ZNyhRFDSMUaTsGQPBRgToosCHuPuZb7VbVtRPt9diUZtY274CBdgTx7bCipKewu8UN9EavtTCHFQN4FXJZTRmrJj16W3ZHjJoSf1H33CA9PyhbSxZ5tGX9SNxD3K14EQuTdYpeqiVaCWbibGJHC3w7ZHxGAKnnBDKZhREH2bzQGhFKc9N4vMstZ36R8mxkU3K1wFjKQCxUvPc7fxxveQ24hFUTNaVL6CYLvRrG29GwHNbcFPBeBBQxGTfSHgp4ciU6DFwtfwQTJL1aJwatPwvvPBydm4YZ1PHhZfN4Lw3UtemPW1kL7vpGmNm3H4tf8xZrZDvjYzRVD62GmpyD2pXvAuBza43SaCjnreCwjMb2NhoUSnYLyY7xNwbHd3QA5sp19u17ti7NuzDuw3Tzx6bcWUaZkh3YbZxrTsYcVNEnwYyFvVfu2a9f74P",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutmMcd9xviSb8h5LLThVBg4B8jFZcCK5jQBDDSUjYw7xj4DJgERvw4R9WrooQwdG8jT6gAdYLn8zs1thKpPmyjTmobLQTQHuygUhuktQ3kW1YhsQ2p1YnPJc4rPXuwBbViCZ9uhCR3wujkwa7UyD2APgrW26cmtRFhXPKCLsKqb57mtHsxYtcyMQjPXHgMCxC7hP7BjHBew7SbYbx5e573a9aHvtH1ziH37cTLttJw2VbXA2guwwrYFGgE4UEbVT9p2XcKsZGFLh9TUzR2gxLV2RzAU2ucxQD23VtBGxaGc9TomVgWsmQQ2qvMhp6v8muHne8gEsodT5WUHxTvP2y3D7425akkzHa5qDWrFGCtbzh8uqoHPXYL43TpDcvMe8WzMuK6otAt2bsMCmxu6ZPaxpDaZfAQtUXCVnhYCPYERBimyBs22bTLrkwDJ3ezY",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutmxDayQ9Y7YSrTd5JwWboK9rHuMtaBvPuJtZLffDWMExrac1svCXism8MVo8NEwSbKiMH2ZMwqh8gSBoFvKC5bhibnjKgeEvz9kazdWACkke2DnMhqrE8DNzB9YvKiq4R4qbG4LdDX4Lg7oULQgYqinjWHNWCfufAxnCSKVHgiykJN5LtHmmXcpsGa66R1zGadWPnBYFrs1mesisMwFk6JSuoEDLsaDFpnvkqh2Gg9ZcMDLrp2VSuc2etcqQB4pE9L8oXpmtzUr9JB6U1My1JEFQRmY13oypFHifgeuqFwe3mo7qSkNA4EwD9CynAfCdHrJvgAhqKEFfKjiFgZYG4FC3xVACS4ybAp7noEU4MqpAGEjussQGQo6o8hxRhuvMsaakaWsbiHh8k3UAbTvNcV5LC2b5weNkW6WwZTyi6tyNifMirr2V9qdt7vVXjF",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutnSrSz7F9dRY239WK9T1SGCxGG1oJUAqwYvoYJ51hk1Z6tsSHevD4wMGv328idz8LA9S89Yhd4ckuVH5PQqcBxTBtqbZCEzxfmbdVvCGLngQ19KHdVoCefomrHmG1Rc7BtCYj7ba8dwCjGn8x26MrCZmpq1N1N5ALm2HDKAGHQgRGqME4UxXw4cVJtgL58mhLiSrr1qxrtGh5TxU6GMHFHaiVoi3FreQkNtXmZBu11bvudqRLDGB7Yy9bZGY841dJKzECQwEf1L2epQJ4KGFdbMRtsRGGSg71m2FD6YsuhUpPH4jhQxL5kVb7atnURhGFZfeGFEHDRco9xZvbU15jUf1392n9Ap8Zf35ZFs7mXvDKP4wYTvzXYou1mYPAQpmJw6Aakcd8JSq1v2idciXjsZV2zX9xihxeJ2x8XNrtQET7HAyKrY2oVwHkVMGHN",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkuto9pSrbTW4dkpS91AFEhM7CyKeCUkcCYFcpTxwurG7XswPPerUjbMvadB7MtvHDGdxYRkfL6NpY6ofxLwb9ufcYzy2R3qzjdpV48XsisB2c1Si7MgPcL7Yb13iPc6qF3mCFekqZZXVVTx43HWitdsioqQbgtakkTg7EpJVcJS7xHPqCE2jyHQb6rjmyyDrbANpGjd3aKXw1jsYWcqri6pXEPyQd3hzux3C6MECRBZ44VNSTSef5TWLXb3f8cYka6eJXgzmxnsv3UQE6TerikbTshgq7mEnMtm2BTd7QkHjgHRYQqYes1hTyXryoPBTADH3mBTpBWCxbk7FmcYLD2ywKfEXkVZCuP5BjBmH8oAeP3XC5ysPRXAXiMAQYWfLVrFHRtkPCUgefEcYDimMDAmnqNhVDVCuP88c7jmP7vMbb9ir3j71na6kDgWdG5Bn",
 "CoaCsVSmrGnEoTzjTqmtMbmqJRVsLkutoj2yoMmssBCXaJamyEqxxn1qG8qUwxZ4BEstABFNTSxmUGMCsxnyh3FS37ZcZKM8XrXTitZXupzqMHXXTcQ4kzyPa975j1eX3Whdedtz2NHbac8id4S5eGuSr2Ej53LLBq2u9zwucD71MpD1YQTKDC4jS8qXQect1nAtSUW8riyMiok5yjDaNxiiDEnRFaBsoMhV2w16DWNXpygqSc9nzhqRdKBG2uJjZcsd9cJTCgaEiUaFSe1GuAya1RuDCwbgkAqfbY4pBLjuu7Fzf482EdD4RURUwCQJjiBRAHBaRmuuauezdN4ydyoVHtTgAzzPFfcBkoTS5dY5yH5SeSuQn6tY2TufEtQkPC2UtejBrsh3gKJEbkrsDEPqHapGMdNFXggmLn5tvdUcyj7SGKTnTGCruZiaMdgo1bswFJtCvDRq41rWJKgZhr9Gc9yf6",
 "3Qf3pVdBGy8PrWGF5wPMD9JF5vazUKS1NfRwYrYpfYf5TTxg5Qpw4MjDf3y5Gr2xZgY8JrsH9xZ6v12ZJquqdccRQ3WJFfS7Cb4956DjSMSVVAhqsice8ybpGqoXrNbLaUtgNGP6wUxRfBDMVSztFPgvqqy6SSVBLeqt7smfL15W8r65S2nSXSxDR3rxPEpbA79ZfygGuSpyFDmLnWxxHsEU3R9tFHwNksazfAhSq4JwwaDsfZH6XYZuz5tdjdSRD1Cnv53HjvgC9bCgNE7dqBuqdU218ztD4QAG8mdmqZ6ig6FbngKHifaD2EL2Fty5BSu29GC1g2gEA8AWsXUhdvK6xf9ACVY7wgH1KgznYziMcxKm4Z194V4wisMXeVARfSGrBesPdDRizvUhon5i3dWmTYDM6M33NyCUyzP9FpGLRmHtKzTFuKqH2XvHSAX4WtSmuBFxKX5EHbJPdsh9aeyTyJvjjZNY3LrgAniKAi4TiTixAMVb7wWsohXXjCRs2WvvkwDipoHXNrS1GnVmEyt9bN94exY8RArALH7AqTrEio3b2ig6EqJuwEnpS3yEGwsNPTFuBBC54zdtND8ibnvB5ULaWo8xcA1AAgy7v6HTf83xnJzSBU7onBdghyN2Z64VAjfjtSyhDjAmfCXcCFwY1cauUbfEELyWYv5zMPdZe3rvahdKzFAh44Kwkppfc2U6pfmauF9HsZLh3gDmLsBuUYQQ6j96mjxrQL82wPswFM1UvM5dDiFBVxW1N1NzUgxmaWRZCVgHZCFHcnebD5kjjhoFeWcRFxCv5Qe5F2q2JsdZ6Tc9aBeU5hG"
]
```

Expected: `<<41, 239, 48, 219, 93, 72, 102, 85, 106, 33, 114, 58, 0, 139, 44, 71, 208, 255,161, 148, 145, 222, 43, 110, 217, 34, 160, 122, 138, 217, 43, 121>>`

Got: `<<213, 24, 114, 99, 187, 226, 196, 28, 2, 216, 183, 53, 195, 53, 154, 61, 251,180, 131, 42, 123, 245, 168, 32, 39, 216, 220, 82, 93, 15, 206, 122>>`

**Comment by thepiwo** [comment]
Do you have any idea, why this might be incorrect?

**Comment by cytadela8** [comment]
Closed by #655
